### ISSUE TITLE
"Tables" in code

### ISSUE THREAD

**Issue by TonyValenti** [dotnet/roslyn#36859]
Hi All,
This is really more of a C# formatting question rather than a C# language question.  If this isn't the right place to discuss this, I'll be happy to move the question to the recommended location.  If you are reading this via email, I'm sure the formatting won't be right.  Please make sure to look at this through the web.

Often times when I'm writing code, I have "tables" where I define things.  A table might look like this:
````
            var T1 = new Dictionary<BillState, int>() {
                [BillState.Awaiting_Approval     ] = 0001,
                [BillState.Awaiting_Payment      ] = 0002,
                [BillState.Deleted               ] = 0010,
                [BillState.Draft                 ] = 0310,
                [BillState.Paid                  ] = 1234,
                [BillState.Void                  ] = 9999,
            };
            
````
or this:
````
var T2 = new List<CustomObject>() {
   {0001, true , Enum.Value1},
   {1001, false, Enum.Value2},
   {3001, true , Enum.Value3},
};
````

You'll notice that I'm trying to use formatting to make the code much more readable and easier to edit (I can just press the insert key on my keyboard and type over a value or use control+alt+shift to select a region and type over it).

The challenge is, when I press semicolon, my "table" gets formatted into something much less readable like this:
````
var T1 = new Dictionary<BillState, int>() {
                [BillState.Awaiting_Approval] = 0001,
                [BillState.Awaiting_Payment] = 0002,
                [BillState.Deleted] = 0010,
                [BillState.Draft] = 0310,
                [BillState.Paid] = 1234,
                [BillState.Void] = 9999,
            };
````

is there a way to tell VS/C# to respect whitespace in table-like structures?

**Comment by CyrusNajmabadi** [comment]
Probably dupe of https://github.com/dotnet/roslyn/issues/31691 @sharwell

**Comment by sharwell** [comment]
Duplicate of #28729

**Comment by sharwell** [comment]
This isn't a duplicate of other issues, but is another formatting request which is blocked on that API.
